### PR TITLE
Raise error for `pusher-v4` environment when `mujoco>=3`

### PR DESCRIPTION
**Pull request by Kallinteris-Andreas**
# Description

https://github.com/Farama-Foundation/Gymnasium/issues/950 noticed that pusher-v4 had collision issues which was isolated to a bug fixed in mujoco==3.0.0
This collision issue has a major implement on performance, therefore, raising an error for pusher-v4 when mujoco>=3
The issue will be fixed in pusher-v5

### CODE REVIEW

**Comment by Kallinteris-Andreas** [comment]
What do we do for testing?

**Comment by pseudo-rnd-thoughts** [comment]
> What do we do for testing?

Remove from the set of testing environments

We could add a specialist CI case where mujoco < 3 is installed just in case. 
But with v5 fixing the issue, I don't imagine many people will use it. 

Is there anyway that we can check that a similar bug doesn't exist in other environments?

File: gymnasium/envs/mujoco/pusher_v4.py [hunk 1]
```diff
@@ -22,7 +23,11 @@ class PusherEnv(MujocoEnv, utils.EzPickle):
     }
 
     def __init__(self, **kwargs):
+        assert (
+            mujoco.__version__ < "3.0.0"
+        ), "`Pusher-v4` is only supported on `mujuco<3`."
```
Comment (pseudo-rnd-thoughts) [inline gymnasium/envs/mujoco/pusher_v4.py:28]: I would add a link to the issue for users to understand why

**Review by pseudo-rnd-thoughts — APPROVED** [review]
@Kallinteris-Andreas Were you planning on anymore changes?

**Comment by Kallinteris-Andreas** [comment]
I have no idea why this is passing CI

**Comment by Kallinteris-Andreas** [comment]
Did not realize you commitef to it, makes sense now

File: tests/envs/mujoco/test_mujoco_v5.py [hunk 2]
```diff
@@ -331,15 +331,17 @@ def test_reward_sum(version: str):
         env_conf("HumanoidStandup", True, False, False, "superset"),
         env_conf("InvertedDoublePendulum", True, True, False, "superset"),
         env_conf("InvertedPendulum", False, True, False, "superset"),
-        env_conf("Pusher", False, True, False, "keys-superset"),
+        env_conf("Pusher", True, True, False, "keys-superset"),  # pusher-v4
```
Comment (pseudo-rnd-thoughts) [inline tests/envs/mujoco/test_mujoco_v5.py:334]: Why was this changed?

File: tests/envs/mujoco/test_mujoco_v5.py [hunk 3]
```diff
@@ -584,17 +586,18 @@ def test_model_object_count(version: str):
...
-    assert env.model.nq == 11
-    assert env.model.nv == 11
-    assert env.model.nu == 7
-    assert env.model.nbody == 13
-    if mujoco.__version__ >= "3.1.2":
-        assert env.model.nbvh == 8
-    assert env.model.njnt == 11
-    assert env.model.ngeom == 21
-    assert env.model.ntendon == 0
+    if not (version == "v4" and mujoco.__version__ >= "3.0.0"):
```
Comment (pseudo-rnd-thoughts) [inline tests/envs/mujoco/test_mujoco_v5.py:589]: I think this is wrong, I think it should be version is v4 and mujoco < 3

**Review by pseudo-rnd-thoughts — CHANGES_REQUESTED** [review]

File: tests/envs/mujoco/test_mujoco_v5.py (hunk 3 above)
Comment (Kallinteris-Andreas) [reply tests/envs/mujoco/test_mujoco_v5.py:589]: Same thing

Comment (pseudo-rnd-thoughts) [reply tests/envs/mujoco/test_mujoco_v5.py:589]: No, as this should run for pusher v4 and mujoco < 3 which this won't
